### PR TITLE
`Programming exercises`: Change default branch from master to main

### PR DESCRIPTION
**Pull request by krusche**
Changes the default branch from `master` to `main` and makes sure to use the configuration in the yml files in the production and test code.

**Note:** Administrators can override the yml configuration with a different default branch and could still fall back to `master` if needed.

#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I adapted multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.
#### Changes affecting Programming Exercises
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student

1. Create a new programming exercise (instructor) and participate (student). Make sure the `main` branch is used
2. Student: Participate in an existing programming exercise (in which master was used) and make sure everything still works
3. Instructor: Import an existing programming exercise (in which master was used) and make sure everything works (also when participating as a student)

Note: this should be tested on both environments (Bitbucket/Bamboo and Gitlab/Jenkins)

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsBuildPlanService.java [hunk 1]
```diff
@@ -216,7 +216,7 @@ public String getBuildPlanKeyFromTestResults(TestResultsDTO testResultsDTO) thro
     }
 
     /**
-     * Copies a build plan to another. And replaces the old reference to the master branch by a reference to the default branch
+     * Copies a build plan to another. And replaces the old reference to the main branch by a reference to the default branch
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsBuildPlanService.java:219]:
```suggestion
     * Copies a build plan to another. And replaces the old reference to the master and main branch by a reference to the default branch
```

File: src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/jobs/JenkinsJobService.java [hunk 2]
```diff
@@ -86,7 +86,7 @@ public FolderJob getFolderJob(String folderName) {
     }
 
     /**
-     * Gets the xml config of the job that is inside a folder and replaces the old reference to the master branch by a reference to the default branch
+     * Gets the xml config of the job that is inside a folder and replaces the old reference to the main branch by a reference to the default branch
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/jobs/JenkinsJobService.java:89]:
```suggestion
     * Gets the xml config of the job that is inside a folder and replaces the old reference to the master and main branch by a reference to the default branch
```

File: src/main/webapp/app/shared/image/session-storage-strategy.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ import { GlobalCacheConfig, ICachePair, IStorageStrategy } from 'ts-cacheable';
  * instead of the localStorage: https://github.com/angelnikolov/ts-cacheable/blob/master/common/LocalStorageStrategy.ts
  */
 export class SessionStorageStrategy extends IStorageStrategy {
-    private masterCacheKey: string = GlobalCacheConfig.globalCacheKey;
+    private mainCacheKey: string = GlobalCacheConfig.globalCacheKey;
```
Comment (Strohgelaender) [inline src/main/webapp/app/shared/image/session-storage-strategy.ts:8]: I don't think the changes in this file were intentional.

Comment (krusche) [reply src/main/webapp/app/shared/image/session-storage-strategy.ts:8]: They actually were intentional. The term master should be avoided completely if possible

File: src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsBuildPlanService.java (hunk 1 above)
Comment (julian-christl) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsBuildPlanService.java:219]:
```suggestion
     * Copies a build plan to another and replaces the old reference to the master and main branch with a reference to the default branch
```

**Review by julian-christl — CHANGES_REQUESTED** [review]
Please handle the two open comments, otherwise the changes look good 👍

**Review by Dominik-Weinzierl — COMMENTED** [review]
I tested the test cases using a local Bitbucket/Bamboo setup:

1. Repository was created with the main branch as expected. 
2. The student was still able to push and edit the task.
3. The main branch was used for the newly imported task (even if the old branch was master).

Important: Build plans and automated tests not yet tested

**Review by julian-christl — APPROVED** [review]
Approve code after resolving the comments. Looks good!

**Review by Strohgelaender — APPROVED** [review]
Tested locally, the main branch gets used as expected.

**Review by b-fein — APPROVED** [review]
Only minimal changes in the wording of the comments since the last review. Re-approve.

**Review by Dominik-Weinzierl — APPROVED** [review]
Also works with Bamboo and automated tests :+1:

**Pull request merged**